packages/ui5-application-writer/src/data/defaults.ts

```diff
@@ -75,19 +80,41 @@ export function mergeUi5(ui5: Partial<UI5>): UI5 {
         framework,
         frameworkUrl: ui5.frameworkUrl ?? defaultFrameworkUrl
     };
-    const typesVersion = parseFloat(merged.localVersion) >= 1.76 ? merged.localVersion : '1.71.18';
+    // typesVersion must be a valid npm semantic version, we know they cannot be null as already validated
+    const localSemVer = semVer.valid((semVer.coerce(merged.localVersion)))!;
+    const typesVersion = semVer.gte(localSemVer, '1.76.0') ? localSemVer : '1.71.18';
```

Comment:
Use constants for `'1.76.0'` and `'1.71.18'` too? I know this PR does not add this code but touches it.